lib/CL/devices/basic/basic.c

```diff
@@ -293,6 +293,10 @@ pocl_basic_init_device_infos(unsigned j, struct _cl_device_id* dev)
 
   dev->device_side_printf = 1;
   dev->printf_buffer_size = 16 * 1024 * 1024;
+  dev->spir_compatible_triple = SPIR_HOST_TARGET_TRIPLE;
+  dev->spir64_datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128";
```

Comment:
SPIR does have a fixed DL, but we need to forcibly change the module DL to device-specific DL before linking, otherwise you get LLVM assertion errors.

Reply:
OK, why it's called spir64 datalayout then? You are forcing it to the target-specific data layout, so it is actually target specific. The above looks like x86-64 basic layout string.

Reply:
Could we ask the datalayout string from the generated bitcode library?